src/transactions/TransactionFrame.cpp

```diff
@@ -357,6 +357,21 @@ TransactionFrame::hasDexOperations() const
     return false;
 }
 
+bool
+TransactionFrame::isSoroban() const
+{
+    return mOperations[0]->isSoroban();
+}
+
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+SorobanResources
+TransactionFrame::sorobanResources() const
+{
+    releaseAssert(isSoroban());
```

Comment:
It should be safer if we throw an exception here and get an `txINTERNAL_ERROR` instead of shutting down. I only use an assert if we can't recover from the error.

Reply:
I treat assertion as 'if this happens it would be an implementation bug' and here we deal with an internal function and it would be nice to get immediate feedback if it's misused.

Reply:
Wouldn't we get feedback with a `txINTERNAL_ERROR`? If you look at the rest of `TransactionFrame`, we throw exceptions (although none of those are internal functions). We use asserts and exceptions throughout our code base so it's not wrong, but I usually err on the side of caution. I'll leave this up to you.

Reply:
Ok, changed to `releaseAssertOrThrow`